src/ai/backend/manager/server.py

```diff
@@ -281,6 +292,83 @@ async def manager_status_ctx(root_ctx: RootContext) -> AsyncIterator[None]:
     yield
 
 
+@actxmgr
+async def leader_election_ctx(root_ctx: RootContext) -> AsyncIterator[None]:
+    configuration = root_ctx.local_config["manager"]["raft-configuration"]
+    raft_id = configuration[root_ctx.pidx]
+    grpc_port = int(raft_id.split(":")[-1])
+    filtered_configuration = tuple(conf for conf in configuration if conf != raft_id)
+
+    root_ctx.node_id = f"{raft_id}:{os.getpid()}"
+    reigns: bool = False
+    global_timer_ids: Final[List[str]] = []
+
+    async def _on_global_timer_created(context: None, source: str, event: GlobalTimerCreatedEvent):
+        if root_ctx.node_id == event.node_id:
+            global_timer_ids.append(event.timer_id)
+            if reigns:
+                await root_ctx.event_producer.produce_event(
+                    GlobalTimerJoinEvent(node_id=root_ctx.node_id, timer_id=event.timer_id)
+                )
+
+    global_timer_created_handler = root_ctx.event_dispatcher.subscribe(
+        GlobalTimerCreatedEvent, None, _on_global_timer_created
+    )
+
+    async def _on_state_changed(next_state: RaftState):
+        nonlocal reigns
+        _log_prefix = f"[pid={os.getpid()}/pidx={root_ctx.pidx}]"
+        match next_state:
+            case RaftState.LEADER:
+                log.debug(f"{_log_prefix} LEADER")
+                reigns = True
+                for timer_id in global_timer_ids:
+                    await root_ctx.event_producer.produce_event(
+                        GlobalTimerJoinEvent(node_id=root_ctx.node_id, timer_id=timer_id)
+                    )
+
+            case RaftState.CANDIDATE:
+                log.debug(f"{_log_prefix} CANDIDATE")
+                reigns = False
+
+            case RaftState.FOLLOWER:
+                log.debug(f"{_log_prefix} FOLLOWER")
+                reigns = False
+                for timer_id in global_timer_ids:
+                    await root_ctx.event_producer.produce_event(
+                        GlobalTimerLeaveEvent(node_id=root_ctx.node_id, timer_id=timer_id)
+                    )
+
+    server = GrpcRaftServer()
+    client = GrpcRaftClient()
+    raft = await Raft.new(
+        raft_id,
+        server=server,
+        client=client,
+        configuration=filtered_configuration,
+        on_state_changed=_on_state_changed,
+    )
+
+    _cleanup_coroutines: List[Coroutine] = []
+
+    _ = (
+        asyncio.create_task(
+            GrpcRaftServer.run(
```

Comment:
I am confused here because `GrpcRaftServer.run()` method is not a classmethod or staticmethod.
https://github.com/lablup/aioraft-ng/blob/main/aioraft/server.py#L28
Is this intended?